src/mixxxmainwindow.cpp

```diff
@@ -398,6 +398,13 @@ void MixxxMainWindow::initialize() {
             &PlayerInfo::currentPlayingTrackChanged,
             this,
             &MixxxMainWindow::slotUpdateWindowTitle);
+
+    // Start Auto DJ if the cmdline arg is passed.
+    if (CmdlineArgs::Instance().getStartAutoDJ()) {
+        qDebug("Enabling Auto DJ from CLI flag.");
+        ControlProxy* pAutoDjEnabled = new ControlProxy("[AutoDJ]", "enabled");
+        pAutoDjEnabled->set(1.0);
```

Comment:
`new` without `delete` is a memory leak in C++. We avoid this pair in new code by using smart pointer.
In this case you can set the CO directly via: 
```suggestion
        ControlObject::set(ConfigKey("[AutoDJ]", "enabled"), 1.0);
```